docs/howto/create-a-release.rst

```diff
@@ -0,0 +1,105 @@
+###########################
+Creating a release for Cylc
+###########################
+
+
+Before you start (First time)
+=============================
+* Create a GPG key for verification on Github: `Instructions <https://help.github.com/en/articles/generating-a-new-gpg-key>`_
+  At step 3 I had to use ``gpg --default-key rsa4096 --gen-key``.
+
+* `Share this key locally with git, as well as Github: <https://help.github.com/en/articles/telling-git-about-your-signing-key>`_
+
+* Create a PyPI account for yourself if you don't already have one.
+  `Sign up page <https://www.google.com/url?sa=t&rct=j&q=&esrc=s&source=web&cd=1&cad=rja&uact=8&ved=2ahUKEwjouIq44MrjAhUKTsAKHeeoCyQQFjAAegQIAxAB&url=https%3A%2F%2Fpypi.org%2Faccount%2Fregister%2F&usg=AOvVaw2ZnHHcXEbkC1c5UJPPMg5a>`_
+  Get an owner to add you as a maintainer or owner.
+
+
+Create pre-release Pull requests
+================================
+
+* Ensure that version numbers are consistent in the following files:
+  * ``cylc-flow/cylc/flow/__init__``, variable ``__version__ = "8.?.?"``.
+  * ``cylc-flow/setup.py``
+  * ``cylc-uiserver/setup.py``
+* Examine Pull Requests made since the last release for:
+  * cylc-flow
+  * cylc-conda
+  * cylc-ui
+  * cylc-uiserver
+* Add to ``CHANGES.md`` text in each repository.
+* Check the ``CONTRIBUTING.md``. Ensure that all contributors are listed.
+  You can do this using Github, or locally with ``git shortlog -sn``
+* Get these PRs reviewed and merged to master.
+
+
+Test the build
+==============
+
+* Check locally that you have the updated version of master. Ensure that you
+  use git fetch as well as git pull to get the tags.
+* Use ``git log`` to ensure that you have the correct version and tag.
+* Check that you can buid your distribution locally:
+
+.. code-block:: bash
+
+  # deliberately leave your repository
+  # so that the build is a more thorough check
+  cd ~/
+
+  # This seems to be really slow
```

Comment:
"   # This seems to be really slow" maybe append " as this will build the wheel as well as the source distributions" (bdist => binary distribution, sdist => source distribution, wheel is a ZIP with a different name like JARs, FWIW)